language/tools/cost_synthesis/src/module_generator.rs

```diff
@@ -114,6 +114,21 @@ impl ModuleBuilder {
         let function_sig_offset = self.module.function_signatures.len();
         self.module.string_pool.append(&mut names);
 
+        self.module.function_handles = sigs
+            .iter()
+            .enumerate()
+            .map(|(i, _)| FunctionHandle {
+                name: StringPoolIndex::new((i + offset) as u16),
```

Comment:
Wrap around with truncation to u16. If intended, make this explicit by using the appropriate library routines.

Reply:
This cost synthesis code has only been reordered; the functionality has not been changed. I will make a note to go over that crate with MIRAI if time permits.